roombasim/pittras/ai/callback_waypoint_demo_controller.py

```diff
@@ -0,0 +1,48 @@
+'''
+callback_waypoint_demo_controller.py
+'''
+
+from roombasim.ai import Controller
+
+class CallbackWaypointDemoController(Controller):
+    '''
+    A demo controller that uses the XYZTranslationTask to move
+    the drone to four waypoints in a circular motion.
+    '''
+
+    def setup(self):
+        # set the initial target
+        self.task_controller.switch_task(
+            'XYZTranslationTask',
+            callback = self.waypoint_callback,
+            target = [5,5,2]
+        )
+
+        self.waypoints = [
+            [5,5,2],
+            [7.5,3,1],
+            [17,5,1],
+            [17,6,1],
+            [19,19,2],
+            [15,15,2],
+            [5,15,1]
+        ]
+
+        self.selected = 0
```

Comment:
Also another naming convention thing. Typically we designate private variables like `self._selected`, with the `_`. This appears to be used only in this class, so I am assuming it is private (please correct me if I am wrong). Did we want to apply this naming convention to the Sim as a whole, as it makes debugging easier (in my experience)? Also, the same convention would apply to methods as well.

Reply:
Yes, it is private but the controller internal code tends to be the highest level code in the whole sim. It is as if the whole sim is the framework and the controller code is the high-level script that uses that framework to actually develop the ai. Therefore, I'm not too concerned about designating privative variables here.

Reply:
From my understanding, public variables are ones that should be accessed through getters and settings and should not be directly referenced outside the class. Though as stated by @hgarrereyn, the controller code is the high-level script that calls into the AI framework, so I don't think the variables need to be prefixed with an underscore.